generators/client/templates/angular/src/main/webapp/app/blocks/interceptor/_notification.interceptor.ts

```diff
@@ -31,22 +32,31 @@ export class NotificationInterceptor extends JhiHttpInterceptor {
     }
 
     responseIntercept(observable: Observable<Response>): Observable<Response> {
-        return <Observable<Response>> observable.catch((error) => {
-            const arr = Array.from(error.headers._headers);
+        return observable.map((response: Response) => {
             const headers = [];
-            let i;
-            for (i = 0; i < arr.length; i++) {
-                if (arr[i][0].endsWith('app-alert') || arr[i][0].endsWith('app-params')) {
-                    headers.push(arr[i][0]);
+            response.headers.forEach((value, name) => {
+                if (name.toLowerCase().endsWith('app-alert') || name.toLowerCase().endsWith('app-params')) {
+                    headers.push(name);
+                }
+            });
+            if (headers.length > 1) {
+                headers.sort();
+                const alertKey = response.headers.get(headers[ 0 ]);
+                if (typeof alertKey === 'string') {
+                    const alertService: JhiAlertService = this.injector.get(JhiAlertService);
```

Comment:
may be we should get the service and set it to a local variable in constructor instead of calling injector on every alert